File: Examples/CallGraphDiff/Kotlin/Before/Services.kt
/** Places an order by charging payment and persisting it. */
class OrderService(private val payment: PaymentService, private val repository: OrderRepository) {
    fun place(order: Order) {
        payment.charge(order)
    }
}

class PaymentService {
    fun charge(order: Order) {}
}

class OrderRepository {
    fun save(order: Order) {}
}

File: Examples/CallGraphDiff/Dart/Before/services.dart
/// Places an order by charging payment and persisting it.
class OrderService {
  final PaymentService payment;
  final OrderRepository repository;

  OrderService(this.payment, this.repository);

  void place(Order order) {
    payment.charge(order);
  }
}

class PaymentService {
  void charge(Order order) {}
}

class OrderRepository {
  void save(Order order) {}
}
